var/spack/repos/builtin/packages/netlib-lapack/package.py

```diff
@@ -182,82 +184,53 @@ def headers(self):
-        else:
-            args.extend(["-DLAPACKE:BOOL=OFF", "-DLAPACKE_WITH_TMG:BOOL=OFF"])
-
-        if self.spec.satisfies("@3.6.0:"):
-            args.append("-DCBLAS=ON")  # always build CBLAS
+        args = [
+            self.define_from_variant("BUILD_SHARED_LIBS", "shared"),
+            self.define_from_variant("LAPACKE", "lapacke"),
+            self.define_from_variant("LAPACKE_WITH_TMG", "lapacke"),
+            self.define("CBLAS", self.spec.satisfies("@3.6.0:")),
+        ]
 
         if self.spec.satisfies("%intel"):
             # Intel compiler finds serious syntax issues when trying to
             # build CBLAS and LapackE
-            args.extend(["-DCBLAS=OFF", "-DLAPACKE:BOOL=OFF"])
+            args.extend([self.define("CBLAS", False), self.define("LAPACKE", False)])
 
         if self.spec.satisfies("%xl") or self.spec.satisfies("%xl_r"):
             # use F77 compiler if IBM XL
             args.extend(
                 [
-                    "-DCMAKE_Fortran_COMPILER=" + self.compiler.f77,
-                    "-DCMAKE_Fortran_FLAGS="
-                    + (" ".join(self.spec.compiler_flags["fflags"]))
-                    + " -O3 -qnohot",
+                    self.define("CMAKE_Fortran_COMPILER", self.compiler.f77),
+                    self.define(
+                        "CMAKE_Fortran_FLAGS",
+                        " ".join(self.spec.compiler_flags["fflags"]) + " -O3 -qnohot",
+                    ),
                 ]
             )
 
         # deprecated routines are commonly needed by, for example, suitesparse
         # Note that OpenBLAS spack is built with deprecated routines
-        args.append("-DBUILD_DEPRECATED:BOOL=ON")
+        args.append(self.define("BUILD_DEPRECATED", True))
 
         if self.spec.satisfies("+external-blas"):
             args.extend(
                 [
-                    "-DUSE_OPTIMIZED_BLAS:BOOL=ON",
-                    "-DBLAS_LIBRARIES:PATH=" + self.spec["blas"].libs.joined(";"),
+                    self.define("USE_OPTIMIZED_BLAS", True),
+                    self.define("BLAS_LIBRARIES:PATH", self.spec["blas"].libs.joined(";")),
                 ]
             )
 
         if self.spec.satisfies("+xblas"):
             args.extend(
                 [
-                    "-DXBLAS_INCLUDE_DIR=" + self.spec["netlib-xblas"].prefix.include,
-                    "-DXBLAS_LIBRARY=" + self.spec["netlib-xblas"].libs.joined(";"),
+                    self.define("XBLAS_INCLUDE_DIR", self.spec["netlib-xblas"].prefix.include),
+                    self.define("XBLAS_LIBRARY", self.spec["netlib-xblas"].libs.joined(";")),
                 ]
             )
 
-        args.append("-DBUILD_TESTING:BOOL=" + ("ON" if self.run_tests else "OFF"))
+        args.append(self.define("BUILD_TESTING", self.pkg.run_tests))
 
         return args
-
-    # Build, install, and check both static and shared versions of the
-    # libraries when +shared
```

Comment:
Changing the installation behavior of netlib-lapack definitely seems out-of-place for this PR: but now that I know it isn't a big deal. IMO though this would be good to separate into a different commit as part of this PR.